contracts/src/systems/raid.cairo

```diff
@@ -0,0 +1,128 @@
+        let adversary_entity_live_coord: Coord = adversary_entity_travel.live_location(
+            adversary_entity_position.into(), 
+            ts
+        );
+        
+
+        let live_distance_between_entities 
+            = raiding_entity_live_coord.measure_distance(adversary_entity_live_coord);
+        
+        // todo@credence move hardcoded distance to config
+        assert(live_distance_between_entities <= 1, 'too far to attack');
+
+
+
+        let adversary_entity_health = get!(ctx.world, adversary_entity_id, Health);
+        assert(adversary_entity_health.balance > 0, 'advesary can not be raided');
+      
+        let raiding_entity_capacity = get!(ctx.world, raiding_entity_id, Capacity);
+        assert(raiding_entity_capacity.weight_gram > 0, 'raiding entity has no capacity');
+
+
+        let mut raiding_entity_attack = get!(ctx.world, raiding_entity_id, Attack);
+        assert(raiding_entity_attack.can_attack(), 'raiding entity cant attack');
+              
+        if raiding_entity_attack.launch() { 
+            // successful attack
+
+            let raiding_entity_quantity = get!(ctx.world, raiding_entity_id, Quantity);
+            let raiding_entity_total_capacity = raiding_entity_capacity.weight_gram * raiding_entity_quantity.get_value();
+
+            let mut resource_total_weight = 0;
+
+            loop {
+                match resource_types.pop_front() {
+                    Option::Some(resource_type) => {
+                        let adversary_resource = get!(ctx.world, adversary_entity_id, Resource);
+
+                        resource_total_weight += WeightConfigImpl::get_weight(
+                                ctx.world, *resource_type, adversary_resource.balance
+                            );
+
+                        if resource_total_weight < raiding_entity_total_capacity {
+                            let raiding_entity_resource = get!(ctx.world, raiding_entity_id, Resource);
+                            set!(ctx.world, (
+                                Resource {
+                                    entity_id: raiding_entity_resource.entity_id,
+                                    resource_type: raiding_entity_resource.resource_type,
+                                    balance: raiding_entity_resource.balance + adversary_resource.balance
+                                },
+                                Resource {
+                                    entity_id: adversary_entity_id,
+                                    resource_type: *resource_type,
+                                    balance: 0
+                                }
+                            ));
+                        }    
+                    },
+
+                    Option::None => {
+                        break;
+                    }
+                };    
+            };
+
+            
+            // todo@credence move hardcoded damage to config
+            // random damage from 1 - 50 
```

Comment:
not sure if we need randomness here, maybe just removing the attack value from the health should suffice for now. We could check with fomentador to make something more fun